app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java

```diff
@@ -0,0 +1,97 @@
+package org.pesmypetcare.mypetcare.activities.fragments;
+
+import android.view.View;
+
+import androidx.test.espresso.UiController;
+import androidx.test.espresso.ViewAction;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+import androidx.test.rule.ActivityTestRule;
+
+import com.google.android.material.button.MaterialButton;
+
+import org.hamcrest.Matcher;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pesmypetcare.mypetcare.R;
+import org.pesmypetcare.mypetcare.activities.MainActivity;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.pressBack;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isAssignableFrom;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class TestRegisterPetFragment {
+    @Rule
+    public ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class);
+
+    @BeforeClass
+    public static void setUpClass() {
+        MainActivity.setEnableLoginActivity(false);
+    }
+
+    @Test
+    public void shouldRegisterNewPet() {
+        onView(withId(R.id.flAddPet)).perform(click());
+        onView(withId(R.id.inputPetName)).check(matches(isDisplayed()));
+
+        onView(withId(R.id.inputPetName)).perform(typeText("Linux"), closeSoftKeyboard());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "Linux" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)